rust/src/delta_datafusion.rs

```diff
@@ -147,12 +153,56 @@ impl TableProvider for delta::DeltaTable {
+                                    max_value: new_stats.max_values.get(field.get_name()).and_then(
+                                        |x| {
+                                            let old_stats = stats.clone();
+                                            let max_value = to_scalar_value(x.as_value()?);
+
+                                            match max_value {
+                                                Some(max_value) => match old_stats.max_value {
+                                                    Some(old_max_value) => {
+                                                        if left_larger_than_right(
+                                                            old_max_value.clone(),
+                                                            max_value.clone(),
+                                                        ) {
+                                                            Some(old_max_value)
+                                                        } else {
+                                                            Some(max_value)
+                                                        }
+                                                    }
+                                                    _ => Some(max_value),
+                                                },
+                                                _ => old_stats.max_value,
+                                            }
+                                        },
+                                    ),
+                                    min_value: new_stats.min_values.get(field.get_name()).and_then(
+                                        |x| {
+                                            let old_stats = stats.clone();
+                                            let min_value = to_scalar_value(x.as_value()?);
+
+                                            match min_value {
+                                                Some(min_value) => match old_stats.min_value {
+                                                    Some(old_min_value) => {
+                                                        if left_larger_than_right(
+                                                            min_value.clone(),
+                                                            old_min_value.clone(),
+                                                        ) {
+                                                            Some(old_min_value)
+                                                        } else {
+                                                            Some(min_value)
+                                                        }
+                                                    }
+                                                    _ => Some(min_value),
+                                                },
+                                                _ => old_stats.min_value,
+                                            }
```

Comment:
nitpick, you could reduce nested level and make the code more readable by matching on two values at the same time:

```suggestion
                                            match (min_value, old_stats.min_value) {
                                                (Some(min_value), Some(old_min_value)) => {
                                                    if left_larger_than_right(
                                                        min_value.clone(),
                                                        old_min_value.clone(),
                                                    ) {
                                                        Some(old_min_value)
                                                    } else {
                                                        Some(min_value)
                                                    }
                                                }
                                                (Some(min_value), None) => Some(min_value),
                                                (None, old) => old,
                                            }
```